worlds/lingo/Options.py

```diff
@@ -0,0 +1,104 @@
+from typing import Dict, Union
+from BaseClasses import MultiWorld
+from Options import Toggle, Option, Choice, DefaultOnToggle, Range
+
+
+class ShuffleDoors(Choice):
+    """If on, opening doors will require their respective "keys".
+    In "simple", doors are sorted into logical groups, which are all opened by receiving an item.
+    In "complex", the items are much more granular, and will usually only open a single door each."""
+    display_name = "Shuffle Doors"
+    option_none = 0
+    option_simple = 1
+    option_complex = 2
+
+
+class ProgressiveOrangeTower(DefaultOnToggle):
+    """When "Shuffle Doors" is on, this setting governs the manner in which the Orange Tower floors open up.
+    If off, there is an item for each floor of the tower, and each floor's item is the only one needed to access that floor.
+    If on, there are six progressive items, which open up the tower from the bottom floor upward.
+    """
+    display_name = "Progressive Orange Tower"
+
+
+class ReduceChecks(Toggle):
+    """When "Shuffle Doors" is off, there are far more location checks than there are items.
+    Enabling reduce checks will remove many of the locations that are associated with opening doors.
+    This option is ignored if "Shuffle Doors" is on."""
+    display_name = "Reduce Checks"
+
+
+class ShuffleColors(Toggle):
+    """If on, an item is added to the pool for every puzzle color (besides White).
+    You will need to unlock the requisite colors in order to be able to solve puzzles of that color."""
+    display_name = "Shuffle Colors"
+
+
+class ShufflePanels(Choice):
+    """If on, the puzzles on each panel are randomized.
+    On "rearrange", the puzzles are the same as the ones in the base game, but are placed in different areas."""
+    display_name = "Shuffle Panels"
+    option_none = 0
+    option_rearrange = 1
```

Comment:
Guessing more is planned, here? Like Lingo is in the era of only having Darkid, but not Sigma rando?

Reply:
Yep! I have a Sigma-like randomizer in the works, but it will take a little bit before it is ready, and I wanted to get Lingo into main first.